lib/id3c/cli/command/etl/manifest.py

```diff
@@ -111,6 +112,14 @@ def etl_manifest(*, db: DatabaseSession):
             collection_barcode = manifest_record.document.pop("collection", None)
             collection_identifier = find_identifier(db, collection_barcode) if collection_barcode else None
 
+            sample_origin = manifest_record.document["sample_origin"]
+            # Only UW retrospectives do not have collection barcodes, prevent
+            # ingestion of other samples that do not have collection barcode yet
+            if not collection_barcode and not re.match("(uwmc|nwh|hmc)_retro", sample_origin):
+                LOG.info(f"Skipping sample that is not a UW retrospective sample that does not have a collection barcode.")
+                mark_skipped(db, manifest_record.id)
+                continue
```

Comment:
My main concern here is that it's a new addition of very SFS-specific logic to core ID3C, and that's something which we've been trying to make decrease with time.

I wonder about doing this filtering in the `specimen-manifests` repo instead, before `manifest upload`? That feels like the right place and NDJSON streams make such filtering ~easy (which is why we're using NDJSON). Granted, that wouldn't address the existing records in `receiving.manifest`, but presumably those won't be an issue anymore since the correct/full `warehouse.sample` records exist now.

Reply:
Ah, that is true! Can definitely move the filtering up